### PR TITLE
Add reasons to /smite

### PR DESCRIPTION
**Pull request by DUFCLiam**
#955

### CODE REVIEW

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 1]
```diff
@@ -1,22 +1,24 @@
 package me.StevenLawson.TotalFreedomMod.Commands;
 
 import me.StevenLawson.TotalFreedomMod.TFM_Util;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.bukkit.ChatColor;
 import org.bukkit.GameMode;
 import org.bukkit.Location;
-import org.bukkit.World;
```
Comment (OnlineRepz) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Leave it how it is

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 2]
```diff
@@ -25,41 +27,55 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
         player.setGameMode(GameMode.SURVIVAL);
 
-        //Clear inventory:
+        // clear inventory:
         player.getInventory().clear();
 
-        //Strike with lightning effect:
+        // strike with lightning effect:
         final Location targetPos = player.getLocation();
-        final World world = player.getWorld();
```
Comment (OnlineRepz) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Leave how it is

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 3]
```diff
@@ -25,41 +27,55 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
         for (int x = -1; x <= 1; x++)
         {
             for (int z = -1; z <= 1; z++)
             {
-                final Location strike_pos = new Location(world, targetPos.getBlockX() + x, targetPos.getBlockY(), targetPos.getBlockZ() + z);
-                world.strikeLightning(strike_pos);
```
Comment (OnlineRepz) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Line 57 and Line 58 (Deleted) Leave it how it is

Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: @OnlineRepz You are not in any position to command anyone to make changes. I highly suggest you refrain from doing something alike from now on.

**Comment by JeromSar** [comment]
@DUFCLiam The original issue (https://github.com/TotalFreedom/TotalFreedomMod/issues/848), requires you add reasons to other commands (/mute, /glist, /gadmin) too.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 4]
```diff
@@ -1,22 +1,24 @@
...
-@CommandParameters(description = "Someone being a little bitch? Smite them down...", usage = "/<command> [playername]")
+@CommandParameters(description = "Someone being a little bitch? Smite them down...", usage = "/<command> [playername] [reason]")
 public class Command_smite extends TFM_Command
 {
+    private static Object args;
     @Override
     public boolean run(CommandSender sender, Player sender_p, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
-    {
-        if (args.length != 1)
+        {
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Spaces that do not belong by the looks of the Github previewer.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 5]
```diff
@@ -25,41 +27,55 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
 
         if (player == null)
         {
-            playerMsg(TFM_Command.PLAYER_NOT_FOUND);
+            playerMsg(TFM_Command.PLAYER_NOT_FOUND, ChatColor.RED);
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Is it not already defined as being red in TFM_Command?

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 6]
```diff
@@ -25,41 +27,55 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
+            reason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
+        }
 
-    public static void smite(final Player player)
-    {
-        TFM_Util.bcastMsg(player.getName() + " has been a naughty, naughty boy.", ChatColor.RED);
+        TFM_Util.bcastMsg(player.getName() + " has been a smitted for: ", ChatColor.RED);
 
-        //Deop
+        // deop
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Pointless and irritating change.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 7]
```diff
@@ -25,41 +27,55 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
         player.setOp(false);
 
-        //Set gamemode to survival:
+        // set gamemode to survival:
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Pointless and irritating change.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 8]
```diff
@@ -25,41 +27,55 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
         player.setGameMode(GameMode.SURVIVAL);
 
-        //Clear inventory:
+        // clear inventory:
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Pointless and irritating change.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 9]
```diff
@@ -25,41 +27,55 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
         player.getInventory().clear();
 
-        //Strike with lightning effect:
+        // strike with lightning effect:
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Pointless and irritating change.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java (hunk 3 above)
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: If this line is being removed you may as well remove the surrounding code that is getting the location, however I do not feel it should be removed as now there will be no lightning strike.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java (hunk 1 above)
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: @OnlineRepz Actually, the way that it has been done (If it works, I have not tested it) would be a cleaner way to do it.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java (hunk 9 above)
Comment (Adamx120) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: I aggre with wild

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java (hunk 8 above)
Comment (Adamx120) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: I aggre with Wild

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java (hunk 7 above)
Comment (Adamx120) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: I aggre with Wild

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java (hunk 6 above)
Comment (Adamx120) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Aggring with Wild

**Comment by Wild1145** [comment]
@JeromSar Add inactive tag please.

**Pull request closed without merging**